R/get_timescale_data.R

```diff
@@ -0,0 +1,107 @@
+#' Get geological timescale data
+#'
+#' This function takes a name of a geological timescale and returns data for the
+#' timescale. Valid names include those of built-in `data.frames` ([GTS2020()]
+#' and [GTS2012()]) and exact matches to those hosted by Macrostrat (see full
+#' list here: <https://macrostrat.org/api/defs/timescales?all>). If a built-in
+#' `data.frame` is specified with `name`, a specific `rank` may also be
+#' specified. Valid ranks include "period", "epoch", "stage", "era", and "eon"
+#' as well as partial matches of those names (e.g., "per" or "st"). If a rank
+#' is supplied, only the intervals of that rank will be returned from the
+#' specified timescale.
+#'
+#' @param name \code{character}. The name of the desired timescale.
+#' @param rank \code{character}. The name of the desired rank (if `name`
+#'   specifies a built-in timescale).
+#' @return A `data.frame` with at least the following columns:
+#'   \item{name}{the names of the time intervals.}
+#'   \item{max_age}{the oldest boundaries of the time intervals, in millions of
+#'     years.}
+#'   \item{min_age}{the youngest boundaries of the time intervals, in millions
+#'     of years.}
+#'   \item{abbr}{either traditional abbreviations of the names of the time
+#'     intervals (if they exist) or custom abbreviations created with R.}
+#'   \item{color}{hex color codes associated with the time intervals (if
+#'     applicable).}
```

Comment:
Is it worth including a details section stating which timescales are available via Macrostrat? I guess in comparison to the GTS timescales we also don't have much information about what these timescales are based on or where they are coming from?

Reply:
I'm not sure what we'd really say about them. We have no control over which ones are there, and as you say, we don't have much information about them beyond what is accessible in the API. We could reach out to the Macrostrat folks and try to get references, but ultimately, this seems like it would be a pretty fragile bit of documentation.